### ISSUE TITLE
Integration of CCL Tracker for analytics

### ISSUE THREAD

**Issue by spMohanty** [geotagx/geotagx-theme#48]

**Comment by spMohanty** [comment]
@supranove : While you are at it, can you please add a note on this issue about the analytics events that have been dealt with, and the ones that are left ?

**Comment by othieno** [comment]
@spMohanty Sorry I missed this. Here's the list of analytics and their implementation status.
- [x] analytics.action.signInButtonClicked
- [ ] analytics.action.goingBackButtonClicked
- [x] analytics.action.invalidClick
- [x] analytics.action.visitCategory
- [x] analytics.action.visitProject
- [x] analytics.action.surveyEvent
- [x] analytics.action.internalLinkClicked
- [x] analytics.action.externalLinkClicked
- [x] analytics.action.shareCategory
- [x] analytics.action.shareProject
- [x] analytics.action.startProject
- [x] analytics.action.startTask
- [x] analytics.action.submitTask
- [x] analytics.action.answerQuestion
- [x] analytics.action.imageZoom
- [x] analytics.action.noImage
- [x] analytics.action.showPreviousQuestion
- [x] analytics.action.showComments
- [x] analytics.action.showImageSource
- [x] analytics.action.showHelp
- [x] analytics.action.startTutorial
- [x] analytics.action.skipTutorial
- [x] analytics.action.completeTutorial
- [x] analytics.action.correctTutorialAnswer
- [x] analytics.action.wrongTutorialAnswer
- [x] analytics.action.tutorialImageZoom
- [x] analytics.action.noTutorialImage
- [x] analytics.action.showPreviousTutorialQuestion
- [x] analytics.action.showTutorialComments
- [x] analytics.action.showTutorialImageSource
- [x] analytics.action.showTutorialHelp

**Comment by spMohanty** [comment]
Wow !! this one looks like its almost complete :D great work :D

**Comment by spMohanty** [comment]
Comments from Jose :

##  Bugs
- When click on community -> user, It is sent to GA an invalidClick.  
  **(Update :: Suppressed invalid click event when clicking user profile on community page, though it will still send the url of the user account which was clicked on, something like `internalLinkClicked:/account/stanfar/`)**
- When submiting an answer (i.e. at the end of the task) I get an invalidClick and the submission is not tracked. We need to track this to count the number of task done by users.  
  **(Update : Now it should send an event like `internalLinkClicked:/#task_response_submitted` when the user submits a task response)**
- When I click on a project…. 3 events are fired: invalidClick:fa fa-play:h...geotagx.org/#project_grid, startProject:wintershelte...ect/wintershelter/newtask, internalLinkClicked:/project/wintershelter/newtask  
  **(UPDATE :: fa-play invalidClick events are suppressed)**
- Skip tutorial and contribute link is not properly tracked. it is firing…. 
  internalLinkClicked:/project/wintershelter/newtask and internalLinkClicked:/project/wintershelter/newtask . We need to track this to understand the impact of the tutorials in the learning process. I.e. we need to know which users are doing the tutorial.  
  **(Note from Mohanty :: Add a second argument to the internalClick handler, which optionally adds a #meta-information to the link to help differentiate between types of internalLinks, say `Skip Tutorial` or `Start Contributing`)**  
  **(UPDATE :: Now we mark SkipTutorial links separately, and you should see something like : `internalLinkClicked:/proj...elter/newtask#SkipTuorial` )**
- Start contributing (the big play button) creates 3 events. invalidClick:fa fa-play b...tagx.org/project/cropid2/, visitProject.... , internalLinkClicked:/project/cropid2/newtask.  
  **(Note from Mohanty :: Should be visitProject event when startContributing button is pressed)**  
  **(UPDATE :: fa-play invalidClick events are suppressed )**
- Task_response_submitted should include the project name.  
  **(UPDATE :: now you should get events like `internalLinkClicked:yamun...l#task_response_submitted` when task responses are submitted)**   
- When I click analyse from the main banner it is fired Startproject, visit project, when I click the play after that…. we have visit project again.  
  @supranove : Can you have a look at this ? This startProject and visitProject is a bit confusing for me. Here are the relevant parts from the discussion with jose about this ::  

```
[30/06/15 4:40:40 pm] joselui: the problem I see..... is that I click analyse and then I click start....
[30/06/15 4:40:46 pm] joselui: it is counting 2 visits
[30/06/15 4:40:55 pm] joselui: it is wrong.... it is 1 visit
[30/06/15 4:41:49 pm] joselui: on the other hand.... I would like to be able to count the people who visit the project but they don't start contributing....
[30/06/15 4:42:34 pm] joselui: it is an interesting behaviour, because it means the project is not interesting... or it is not what the user was expecting....
[30/06/15 4:45:12 pm] joselui: and if it is possible.... to remove it form the analyse button
[30/06/15 4:45:19 pm] joselui: that would be just visit
```
- Tutorial finished need to be tracked.  
  **(UPDATE :: Added a `#CompleteTutorial` marker when the user completes a tutorial and clicks on startContributing button)**  
  InvalidClick will be used to evaluate learning aspect regarding the interface. The analytic data should be carefully debugged. There are many invalidClick being fired at wrong places. 

## Other comments
-  Most of userId in the survey do not have any action on the analytics. Check correlation userId/surveyID
-  After tutorial in the shelter project. “Start Contributing” is not visible. (this comment does not relate to analytics)  
  **(UPDATE : To be done in the `geotagx-project-template` repository, and I will wait till we fix the issue with `pbs` before we push this change)**
  **(UPDATE 2: The Start Contributing button works fine now.)**
- Clicking in my profile fires two events. internalLinkClicked:# and internalLinkClicked:/account/Fernandez/ . First one it will be not understandable by anyone analysing the data. 
  **(Update : Supressed internalLinks pointing to '#' as they do not mean anything from an analytics point of view)**
- Community is tracked as /account/ ... it will be confusing for the person doing the analysis.  
  **(Explanation :: this `URI` is specific to the internals of pybossa, I would avoid changing it as we have been trying to keep the geotagx code as much decouples as possible from the pybossa code)**  
- When visiting a project there are to events fired. internalLink and visitProject. It should be taking into account by the person doing the analysis. 
- I don't see clearly the different between visitProject event and startProject. What information is providing each of the events?  
  **(EXPLANATION :: after looking at the code, I realised that`visitProject` event is released when a user opens the `info` page or the `profile` page of a project, while a `startProject` event is released when a user goes to ANY page of a project. Not sure why this naming convention was used exactly, will let @supranove  explain if there is any specific reason before I make any changes)
  (UPDATE -- supranove: I've revised the events such that visitProject is only fired when a user visits the information page and startProject is fired when the user starts contributing to the project.)**
- Add a surveyStart/surveyComplete event  
  **(Note from Mohanty :: A surveyStart event can be added, but a surveyComplete event is not possible as the surveys are completely independent of geotagx and the limesurvey instance currenly doesnt have any mechanism to let us know if the user indeed submitted a valid survey)**  
  **(UPDATE :: Added a surveyEvent, and you should now be able to see events like `surveyType::INITIAL::OPEN` and `surveyType::INITIAL::CLOSE` when the survey modal opens and closes. Please note that, these events do not communicate if the user actually completed the survey, but they simply tell about if and when the survey modal was opened)**

Basically, we need to know which event I use for counting tasks , and access to projects (is it startproject or visitProject?)  
**(Update : Now you should get an event like `internalLinkClicked:/#task_response_submitted` when the user submits a task response, that can be used to compute the number of tasks submitted by the user, and the `visitProject` event can be used to count the number of project page visits)**
